### DISCUSSION TITLE
Compliance of EN 16931

### DISCUSSION THREAD

**Discussion by marvinvalent** [stephanstapel/ZUGFeRD-csharp#297] [Q&A] [answered]
I could not find an answer to this question in this repository, so I will just ask it here:
Is the Comfort profile (version 2.2) of this repository fully compliant with the ZUGFeRD-standard 2.2 and therefore with EN 16931 or is there still work in progress?
Because I found some fields from the Technical Appendix that are not implemented here (or I didn't look for them correctly).
I also saw that some fields are mentioned in the code with a todo comment. Is this the case for all missing fields?

**Comment by stephanstapel** [comment, marked as answer]
There might be fields missing. The approach of developing the library is to implement as much as is needed by the users. If your software and users need more of the fields in the comfort profile, please add them to the readers, writers and test cases and send in a pr.

**Reply by marvinvalent** [reply]
Thanks for your reply.
So it is not granted that all invoices generated by this code are valid for the German law for electronic invoices that's coming in 2025?

**Reply by stephanstapel** [reply]
:)
No, not at all. The invoices generated by the library are perfectly valid. It is just that nobody ever asked for these optional elements that are missing. It appears they cover some edge cases.
